azurelinuxagent/ga/remoteaccess.py

```diff
@@ -0,0 +1,158 @@
+# Microsoft Azure Linux Agent
+#
+# Copyright Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+
+import datetime
+import glob
+import json
+import operator
+import os
+import os.path
+import pwd
+import random
+import re
+import shutil
+import stat
+import subprocess
+import textwrap
+import time
+import traceback
+import zipfile
+
+import azurelinuxagent.common.conf as conf
+import azurelinuxagent.common.logger as logger
+import azurelinuxagent.common.utils.fileutil as fileutil
+import azurelinuxagent.common.version as version
+import azurelinuxagent.common.protocol.wire
+import azurelinuxagent.common.protocol.metadata as metadata
+
+from datetime import datetime, timedelta
+from pwd import getpwall
+from azurelinuxagent.common.errorstate import ErrorState
+
+from azurelinuxagent.common.event import add_event, WALAEventOperation, elapsed_milliseconds
+from azurelinuxagent.common.exception import ExtensionError, ProtocolError
+from azurelinuxagent.common.future import ustr
+from azurelinuxagent.common.protocol.restapi import ExtHandlerStatus, \
+                                                    ExtensionStatus, \
+                                                    ExtensionSubStatus, \
+                                                    VMStatus, ExtHandler, \
+                                                    get_properties, \
+                                                    set_properties
+from azurelinuxagent.common.protocol.metadata import MetadataProtocol
+from azurelinuxagent.common.utils.cryptutil import CryptUtil
+from azurelinuxagent.common.utils.flexible_version import FlexibleVersion
+from azurelinuxagent.common.utils.processutil import capture_from_process
+from azurelinuxagent.common.protocol import get_protocol_util
+from azurelinuxagent.common.version import AGENT_NAME, CURRENT_VERSION
+from azurelinuxagent.common.osutil import get_osutil
+
+REMOTE_USR_EXPIRATION_FORMAT = "%a, %d %b %Y %H:%M:%S %Z"
+DATE_FORMAT = "%Y-%m-%d"
+TRANSPORT_PRIVATE_CERT = "TransportPrivate.pem"
+REMOTE_ACCESS_ACCOUNT_COMMENT = "JIT Account"
+MAX_TRY_ATTEMPT = 5
+FAILED_ATTEMPT_THROTTLE = 1
+
+def get_remote_access_handler():
```

Comment:
There is a unit test for all the handlers, please add this one.

Reply:
I don't see one.  Do you mean extension handlers? 

---
In reply to: [194025434](https://github.com/Azure/WALinuxAgent/pull/1171#discussion_r194025434) [](ancestors = 194025434)

Reply:
No - TestImportHandler